src/diffusers/image_processor.py

```diff
@@ -0,0 +1,173 @@
+# Copyright 2023 The HuggingFace Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import warnings
+from typing import Union
+
+import numpy as np
+import PIL
+import torch
+from PIL import Image
+
+from .configuration_utils import ConfigMixin, register_to_config
+from .utils import CONFIG_NAME, PIL_INTERPOLATION
+
+
+class VaeImageProcessor(ConfigMixin):
+    """
+    Image Processor for VAE
+
+    Args:
+        do_resize (`bool`, *optional*, defaults to `True`):
+            Whether to resize the image's (height, width) dimensions to the specified `size`. `do_resize` in the
+            `preprocess` method.
+        vae_scale_factor (`int`, *optional*, defaults to `8`):
+            scale factor in VAE, if do_resize is True, the image will be automatically resized to multipls of
+            vae_scale_factor
+        resample (`str`, *optional*, defaults to `lanczos`):
+            Resampling filter to use if resizing the image.
+        do_normalize (`bool`, *optional*, defaults to `True`):
+            Whether to normalize the image to [-1,1]
+    """
+
+    config_name = CONFIG_NAME
+
+    @register_to_config
+    def __init__(
+        self,
+        do_resize: bool = True,
+        vae_scale_factor: int = 8,
+        resample: str = "lanczos",
+        do_normalize: bool = True,
```

Comment:
One question here is if we should include the vae "factor" (I don't know what to call it, I'm referring to the `0.18215` in Stable Diffusion, we also use the name "vae scale factor" for it) as part of the pre-/pos-processing. Related discussion: https://github.com/huggingface/diffusers/pull/1860#issuecomment-1368880891 – This is actually pre-processing.

Reply:
Wouldn't include it as it's more part of the VAE model IMO as is done already and not too keen on duplicating params